api/internal/git/repospec.go

```diff
@@ -109,7 +112,7 @@ const (
 // https://github.com/someOrg/someRepo?ref=someHash, extract
 // the parts.
 func parseGitURL(n string) (
-	host string, orgRepo string, path string, gitRef string, gitSubmodules bool, gitSuff string, gitTimeout time.Duration) {
+	host string, domain string, orgRepo string, path string, gitRef string, gitSubmodules bool, gitSuff string, gitTimeout time.Duration) {
```

Comment:
I've changed `parseGitURL` to return `*RepoSpec` instead of all the components. As of now, I've kept the error checks in `NewRepoSpecFromURL` because the placement seems to facilitate separation of concerns: the `parse` and `peel` functions all supply default values when they encounter unexpected url behavior, and `NewRepoSpecFromURL` acts as the validator. 

If this isn't an issue, I'm happy to add the checks into `parseGitURL`. I tried to re-factor `parseGitURL` so that it won't be too clunky to add the checks.